src/osd/PrimaryLogPG.cc

```diff
@@ -10466,6 +10465,7 @@ void PrimaryLogPG::mark_all_unfound_lost(
 	} // otherwise, just do what we used to do
 	dout(10) << e << dendl;
 	log_entries.push_back(e);
+	missing_loc.recovered(oid);
```

Comment:
this should go in the callback passed to submit_log_entries() below - it clears it from needs_recovery_map, so ops can come in for the object before the delete is performed